ls1intum/Artemis
# Feature/automatic text/store feedback consistency

⚠️ Depends on https://github.com/ls1intum/Athene/pull/18 to be deployed as well!

---




### Checklist

### Motivation and Context


As mentioned in the feature request [#1877 ](https://github.com/ls1intum/Artemis/issues/1877), we want to store feedback conflicts that is found during the assessment of text exercises. 
The conflicts are found with an Athene service. Corresponding PR in Athene https://github.com/ls1intum/Athene/pull/18

### Description

Workflow:

1.  When a text assessment is submitted, an asynchronous call is made to the remote service only if the automatic assessment is enabled and assessed text blocks are already assigned to a cluster. 
2. Feedback consistency is checked in Athene (≈1-2 seconds)
3. Found conflicts are returned and saved 

Changes:
- TextAssessmentConflictService is added to communicate with Athene service
- Added TextAssessmentConflict entity and corresponding repository
- Minor Feedback entity extension
- Added multiple test cases
   - Added TextAssessmentConflictServiceTest to test the communication and response of the Athene
   - Added AutomaticTextAssessmentConflictServiceTest to mock the response of the Athene and to test the database operations

### Steps for Testing

Testing in the test server is not possible since https://github.com/ls1intum/Athene/pull/18 is not deployed yet. So, please test it locally. 

1. Start Athene with this [branch](https://github.com/ls1intum/Athene/tree/feature/feedback-consistency)
2. Log in to Artemis
3. Create a text exercise with automatic assessments are enabled
4. Submit a couple of text submissions (use similar texts so they can be clustered together)
5. Wait for the submission due date so Athene can cluster text blocks (you can check Artemis logs or docker dashboard for Clustering component in Athene to see if clustering is started/done.)
6. Assess multiple submissions (with completely different feedback comments and scores 🙂 )
7. Check the database if the inconsistencies are stored 

### Test Coverage

- AutomaticTextAssessmentConflictService - 91%
- TextAssessmentConflictService - 37% ( connector class - test is skipped because Athene is not updated yet.)
